app/src/main/java/com/gh4a/utils/HttpImageGetter.java

```diff
@@ -538,32 +535,25 @@ private Bitmap renderSvgToBitmap(Resources res, InputStream is) {
                 int docWidth = (int) (svg.getDocumentWidth() * density);
                 int docHeight = (int) (svg.getDocumentHeight() * density);
                 if (docWidth < 0 || docHeight < 0) {
+                    // The image doesn't have a specified width or height, so we try to determine
+                    // its aspect ratio to render it with correct proportions
                     float aspectRatio = svg.getDocumentAspectRatio();
-                    if (aspectRatio > 0) {
-                        float heightForAspect = (float) mMaxWidth / aspectRatio;
-                        float widthForAspect = (float) mMaxHeight * aspectRatio;
-                        if (widthForAspect < heightForAspect) {
-                            docWidth = Math.round(widthForAspect);
-                            docHeight = mMaxHeight;
-                        } else {
-                            docWidth = mMaxWidth;
-                            docHeight = Math.round(heightForAspect);
-                        }
+                    if (aspectRatio != -1) {
```

Comment:
Since the method explicitly returns -1 when the ratio cannot be determined, checking for that specific value looked more intention-revealing to me.
Does the safety concern apply here? I think it would if we did the check against the result of a floating-point operation, but getAspectRatio does a plain `return -1` inside. Am I missing something?

Reply:
I generally try to rely as little as possible on implementation details of 3rd party code, which could change at any time. The `> 0` check basically meant to express/test 'a valid aspect ratio was returned'. If the lib returned e.g. -0.5, we surely do not want to use that value.

Reply:
Makes sense. In this case it wasn't really an implementation detail since the behavior is explicitly documented in the JavaDoc, but I'm perfectly fine with reverting it if you prefer.

Reply:
Reverted this. Let me know about the rest.